components/heroes/Heroes.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect } from 'react';
+import styles from '../../styles/heroes.module.scss';
+import Hero from '../hero/Hero';
+import { useAppDispatch, useAppSelector } from '../../hooks/redux';
+import { useFetchAllHeroesQuery } from '../../services/HeroesService';
+import { setTotalPage } from '../../store/reducers/heroesSlice';
+import { Loader } from '../loader/Loader';
+
+const Heroes: React.FC = () => {
+  const { limit, offset } = useAppSelector((state) => state.heroes);
+  const { searchTerm } = useAppSelector((state) => state.search);
+  const dispatch = useAppDispatch();
+
+  const { isLoading, isError, data } = useFetchAllHeroesQuery({
+    limit: Math.abs(limit),
+    offset,
+    search: searchTerm,
+  });
+
+  const result = data?.data?.results;
+
+  const totalPage = !searchTerm ? data && data.data.total - 1240 : 0;
+
+  useEffect(() => {
+    dispatch(setTotalPage(totalPage));
+  }, [dispatch, totalPage]);
+
+  return (
+    <>
+      {isLoading ? (
+        <Loader />
+      ) : !result || !result.length || isError ? (
+        <h1 style={{ padding: '80px 50px 100px' }}>Heroes not found</h1>
+      ) : (
+        <div className={styles.heroes}>
+          {result.map((hero) => (
+            <Hero key={hero.id} {...hero} />
+          ))}
+        </div>
+      )}
+    </>
```

Comment:
Nested ternary operations are not good point code. According to Best Practices, ternary operations should not have nested conditions. Your code could be rewritten as follows:
```jsx
          <>
            {(() => {
              if (isLoading) {
                return <Loader />;
              }

              if (!result || !result.length || isError) {
                return <h1 style={{ padding: '80px 50px 100px' }}>Heroes not found</h1>;
              }

              return <Hero {...hero} key={hero.id} />;
            })()}
          </>
```

Reply:
Another point. Spread should be placed first. This is necessary in order to avoid redefining single props like `key`.